### PR TITLE
Document mxnet versions that do not work with Horovod

### PR DESCRIPTION
**Pull request by EnricoMi**
There are quite a few MXNet versions that do not work with horovod. This tested all available versions >= 1.4.0 and clearly documents the ones that do not work.

[Build 4800](https://buildkite.com/horovod/horovod/builds/4800) shows which are building and which aren't.
[Build 4808](https://buildkite.com/horovod/horovod/builds/4808) shows the ones that are building also pass the tests.

### CODE REVIEW

File: docs/mxnet.rst [hunk 1]
```diff
@@ -54,4 +54,8 @@ The script below provides a simple skeleton of code block based on the Apache MX
 
 
 
-.. NOTE:: The `known issue <https://github.com/horovod/horovod/issues/884>`__ when running Horovod with MXNet on a Linux system with GCC version 5.X and above has been resolved. Please use MXNet 1.4.1 or later releases with Horovod 0.16.2 or later releases to avoid the GCC incompatibility issue. MXNet 1.4.0 release works with Horovod 0.16.0 and 0.16.1 releases with the GCC incompatibility issue unsolved.
+.. NOTE:: Some MXNet versions do not work with Horovod:
+
+    - MXNet 1.4.0 and before has `GCC incompatibility issues <https://github.com/horovod/horovod/issues/884>`__. Please use MXNet 1.4.1 or later with Horovod 0.16.2 or later to avoid these.
```
Comment (sblotner) [inline docs/mxnet.rst:59]: "before has" --> "earlier have"

Comment (sblotner) [inline docs/mxnet.rst:59]: these --> these incompatibilities.

File: docs/mxnet.rst [hunk 2]
```diff
@@ -54,4 +54,8 @@ The script below provides a simple skeleton of code block based on the Apache MX
...
+    - MXNet 1.5.1, 1.6.0, 1.7.0 and 1.7.0.post1 are missing MKLDNN headers, so they do not work with Horovod. Please use 1.5.1.post0 or 1.6.0.post0 instead.
```
Comment (sblotner) [inline docs/mxnet.rst:60]: Add comma after 1.7.0

File: docs/install.rst [hunk 3]
```diff
@@ -77,11 +77,11 @@ To ensure that Horovod is built with MXNet CPU support enabled:
 
     $ HOROVOD_WITH_MXNET=1 pip install horovod[mxnet]
 
-To ensure that Horovod is built with MXNet GPU support enabled for CUDA 10.2:
+Some MXNet versions do not work with Horovod:
 
-.. code-block:: bash
-
-    $ HOROVOD_WITH_MXNET=1 pip install horovod[mxnet-cu102]
+- MXNet 1.4.0 and before has GCC incompatibility issue. Please use MXNet 1.4.1 or later with Horovod 0.16.2 or later to avoid these.
```
Comment (sblotner) [inline docs/install.rst:82]: "before has" --> "earlier have a"

Comment (sblotner) [inline docs/install.rst:82]: avoid these --> avoid these incompatibilities.

Comment (sblotner) [inline docs/install.rst:82]: Remove "please"

File: docs/mxnet.rst (hunk 1 above)
Comment (sblotner) [inline docs/mxnet.rst:59]: Remove "please"

File: docs/mxnet.rst (hunk 2 above)
Comment (sblotner) [reply docs/mxnet.rst:60]: Remove "please"

File: docs/install.rst [hunk 4]
```diff
@@ -77,11 +77,11 @@ To ensure that Horovod is built with MXNet CPU support enabled:
...
+- MXNet 1.5.1, 1.6.0, 1.7.0 and 1.7.0.post1 are missing MKLDNN headers, so they do not work with Horovod. Please use 1.5.1.post0 or 1.6.0.post0 instead.
```
Comment (sblotner) [inline docs/install.rst:83]: Add comma after 1.7.0
Remove "please"

File: docs/install.rst [hunk 5]
```diff
@@ -77,11 +77,11 @@ To ensure that Horovod is built with MXNet CPU support enabled:
 
     $ HOROVOD_WITH_MXNET=1 pip install horovod[mxnet]
 
-To ensure that Horovod is built with MXNet GPU support enabled for CUDA 10.2:
+Some MXNet versions do not work with Horovod:
 
-.. code-block:: bash
-
-    $ HOROVOD_WITH_MXNET=1 pip install horovod[mxnet-cu102]
```
Comment (sblotner) [inline docs/install.rst:84]: If removing this, can we replace with another step for how to install the correct version of MXNet?

Comment (EnricoMi) [reply docs/install.rst:84]: @tgaddair as I understood you, we do not want to go into this detail, right?

**Comment by EnricoMi** [comment]
@sblotner thanks for the comments, embarrassing to get for comments for three changed lines, but that is what I have asked for :smile:

File: docs/install.rst (hunk 5 above)
Comment (tgaddair) [reply docs/install.rst:84]: I think we can leave the example install command, if we also add these version restrictions to the `mxnet-cu102` extra.

**Review by sblotner — APPROVED** [review]

**Pull request merged**